src/components/WebLicenseCode.vue

```diff
@@ -0,0 +1,225 @@
+            if (process.env.NODE_ENV === 'production') {
+                const fieldsFilled = {}
+                for (const detail in this.attributionDetails) {
+                    fieldsFilled[detail] = this.attributionDetails[detail] !== ''
+                }
+                const copiedLicense = {
+                    license: this.shortName,
+                    // codeType can be either rich-text or html
+                    codeType: this.currentSelection,
+                    fieldsFilled: fieldsFilled
+                }
+                this.$ga.event({
+                    eventCategory: 'Attribution',
+                    eventAction: 'copied',
+                    eventLabel: JSON.stringify(copiedLicense)
+                })
+            }
+            this.$emit('copied', { content: e.text })
+            setTimeout(() => {
+                this.success = false
+            }, 2000)
+            e.clearSelection()
+        },
+        onCopyError(e) {
+            this.$emit('copyFailed')
+            e.clearSelection()
+        },
+        clipboardTarget() {
+            return `#attribution-${this.currentSelection}`
+        }
+    },
+    computed: {
+        ...mapGetters(['shortName', 'fullName', 'iconsList', 'licenseUrl']),
+        attributionDetails() {
+            return this.$store.state.attributionDetails
+        },
+        htmlLicenseParagraph() {
+            const data = generateHTML(this.attributionDetails, this.shortName)
+            const licenseCodeSpan = this.$i18n.t('license-use.richtext.full-text', {
+                workTitle: data.workTitle ? data.workTitle : this.$i18n.t('license-use.richtext.workTitle'),
+                creator: data.creator,
+                licenseName: this.shortName,
+                by: data.creator ? this.$i18n.t('license-use.richtext.by') : ''
+            })
+            return `${data.htmlString}${licenseCodeSpan}${data.licenseIconsLink}</p>`
+        },
+        activeTab: {
+            get() { return this.currentTab },
+            set(val) {
+                if (val !== 2) {
+                    this.currentTab = val
+                } else {
+                    this.currentSelection = this.currentTab === 0 ? 'richtext' : 'html'
+                    const tab = this.currentTab
+                    this.currentTab = 2
+                    this.copyText = this.$t('license-use.copied-label')
+                    setTimeout(() => {
+                        this.currentTab = tab
+                    }, 1)
+                    setTimeout(() => {
+                        this.copyText = 'Copy'
+                    }, 2000)
+                }
+            }
+        }
+    },
+    mounted() {
+        this.clipboard = new Clipboard('.copyBtn')
+        this.clipboard.on('success', this.onCopySuccess)
+        this.clipboard.on('error', this.onCopyError)
+    },
+    destroyed() {
+        this.clipboard.destroy()
+    }
+}
+</script>
+<style lang="scss">
```

Comment:
A lot of the styles here look repeated from PrintLicenseCode. I would relocate this CSS to a separate file, maybe create a .scss file in a new dir `./src/styles/` for this, and import it to both components. Make any changes needed per component in the component file itself.

Reply:
Now that I think of it, it would be good to go through and try to find large chunks of shared styles like this and move them to an external file so as not to repeat ourselves. Moving all pseudo-vocab styles to external files may also be good so there aren't remnants of it left behind after we transition fully over to vocab.